### DISCUSSION TITLE
Adding extra files for context

### DISCUSSION THREAD

**Discussion by bannsec** [community/community#6654] [Copilot Conversations]
It appears that context for the AI is dependent on your current open file. It would be helpful if i could allow the AI to use extra files (for instance, header files, other python modules, etc) to increase it's awareness of the application as a whole and i would assume improve it's ability to auto generate code.

**Comment by ryanjsalva** [comment]
@bannsec, is this something you would want to configure or should Copilot just do it auto-magically?

**Reply by bannsec** [reply]
i think it makes sense to opt-in to that functionality, since you could have a bunch of unrelated code that would probably be better off not being part of the context.

**Comment by Kremnari** [comment]
I would like to see opt-in functionality with this as well.
I'm creating a few libraries utilizing a new programming pattern, and it would be nice to give copilot some code hints as it were

**Comment by demipixel** [comment]
Not sure the feasibility of this any time soon since it exponentially increases the amount of code Copilot would need to parse, but this does seem like the next logical step in the project (whether it be months out or years out to this happening).

A slightly simpler method in the meantime might be considering VSCode's autocomplete—if an autocompleted function has the wrong types, rerun copilot including the type definitions of functions it used incorrectly to see if it will generate a better version.

**Reply by Urethramancer** [reply]
The contexts I'd like to see:
1.File only, like now
2.Folder only, for library projects or internal packages with very limited scope
3.Project-wide, with all the source for context

An option to set that for a workspace would be really nice.

**Reply by demipixel** [reply]
Absolutely—I could even see a (4) for cross project (projects that access each other via REST), but I imagine it's a ways before Codex can handle that much code quickly and actually perform meaningful actions with it.
